ivadomed/scripts/automate_training.py

```diff
@@ -242,17 +251,36 @@ def automate_training(config, param, fixed_split, all_combin, n_iterations=1, ru
                     except OSError as e:
                         print("Error: %s - %s." % (e.filename, e.strerror))
 
-            test_scores = pool.map(test_worker, config_list)
-            test_df = pd.DataFrame(test_scores, columns=['log_directory', 'test_dice'])
-            combined_df = val_df.set_index('log_directory').join(
-                test_df.set_index('log_directory'))
+            test_results = pool.map(test_worker, config_list)
+
+            df_lst = []
+            # Merge all eval df together to have a single excel file
+            for j, result in enumerate(test_results):
+                df = list(result).pop()
+                id = result[0].split("_n=")[0]
+                rows = df.index.values
+                for idx, row in enumerate(rows):
+                    df.rename({row: id + "_" + row}, axis='index', inplace=True)
+                df_lst.append(df)
+                test_results[j] = result[:2]
+
+            # Init or add eval results to dataframe
+            if i != 0:
+                eval_df += eval_df + pd.concat(df_lst, sort=False)
+            else:
+                eval_df = pd.concat(df_lst, sort=False)
+
+            test_df = pd.DataFrame(test_results, columns=['log_directory', 'test_dice'])
+            combined_df = val_df.set_index('log_directory').join(test_df.set_index('log_directory'))
             combined_df = combined_df.reset_index()
 
         else:
             combined_df = val_df
 
         results_df = pd.concat([results_df, combined_df])
         results_df.to_csv("temporary_results.csv")
+        eval_df /= n_iterations
+        eval_df.to_csv("average_eval.csv")
```

Comment:
Not sure I followed correctly: what are the columns / rows of this df at this end?

Reply:
Here is the general look:
![image](https://user-images.githubusercontent.com/49137243/87504290-351e5200-c634-11ea-8136-aeab7d15aa90.png)

Each dataframe has the same format as the one for one training, so rows are each test subj and this for each condition we are testing. Rows are all the metrics :). Glad to make modifications to make it more convenient

Reply:
Yes, that's what I thought --> that's why I got confused by the "/= n_iterations" --> how is it used?

Reply:
And is there a reference to the log_directory_idXX in the df? could be handy too!

Reply:
n_iterations is to compute the average (before that it's the sum) :3 . For the reference, I put the name of the log_directory as prefix for each subject (row names), is this what you mean?

Reply:
Oooo got you: there is two df involved: the one where eval_df are appended (ie screenshot above) and the final df (average_eval.csv) which is the average of each metric. All guccci